addons/mail/static/src/js/mail_activity.js

```diff
@@ -0,0 +1,75 @@
+odoo.define('mail.ActivityLog', function (require) {
+"use strict";
+
+var core = require('web.core');
+var Model = require("web.Model");
+var Widget = require('web.Widget');
+
+var QWeb = core.qweb;
+
+// -----------------------------------------------------------------------------
+// Activity Log
+//
+// Integrated with Chatter
+// -----------------------------------------------------------------------------
+var ActivityLog = Widget.extend({
```

Comment:
Since most of the method from this object are used in KanbanActivity widget, I think we should have a different structure like
```
var AbstractActivityLog = Widget.extend({
    init: function(res_model, res_id){...},
    ... // fetching method returning deferred, ...
         // this objct can trigger event catch by subclass and do usefull stuff with them
});

var ActivityLogList = AbstractActivityLog.extend({
    template = 'mail.ActivityLogList',
    ... // All methods and events relative to the chatter part
});

var KanbanActivity = AbstractActivityLog.extend({
    template = 'mail.KanbanActivity',
    ... // All methods and events for kanban widget
});
```

The idea is to split fetching part (models), and the UI (events, rendering, interactions, ....)